### PR TITLE
Scripts/Spells: Implemented druid's Skull Bash

### LINKED ISSUE #19727: Core/Script The spell Skull Bash doesn´t work.
**Description:**

The spell Skull Bash(ID:106839) given to Druid Guardian at lv.70 has an issue

Skull Bash: You charge and bash the target´s skull, intercepting spellcasting and preventing any spell in that school from being cast for 4 sec.

**Current behaviour:**

The spell doesn´t work, it doesn´t do anything when you use it 

**Expected behaviour:**

when you use the spell you should charge and bash the target and interrup the spellcasting, the target can´t cast any spell from that school for 4 sec.

**Steps to reproduce the problem:**

1. Create a Druid character
2. Reach lv.70 and choose the Guardian Spec
3. Use the spell Skull Bash in a target while is casting an spell

**Branch(es):**

Master

**TC rev. hash/commit:** 

7c66e7bbc7a401e7a2d8f2bdcead6531678262ea

**TDB version:**  7.2.0+updates

**Operating system:** Windows 10

Description in wowhead http://www.wowhead.com/spell=106839/skull-bash

### PR DESCRIPTION
**Pull request by matanshukry**
**Changes proposed:**

- Implemented druid's skull bash spell

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

Closes #19727

**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

**Comment by offl** [comment]
#19727?

**Comment by illfated** [comment]
Would it be interesting to sort the enum DruidSpells list, or should we ignore it? I notice it got skipped in your Prayer of Mending PR.

**Comment by matanshukry** [comment]
> Would it be interesting to sort the enum DruidSpells list, or should we ignore it? I notice it got skipped in your Prayer of Mending PR.

I'm planning another PR for paladin spells, was gonna sneak the priest enum into that.
I need to fix this one enum anyway, so might as well do it together as well.

**Comment by Shauren** [comment]
This is incorrect (wrong sequencing and wrong spell ids)
Using 106839 immediately triggers two spells: 221514 (charge) and 93985 (interrupt). Interrupt does not happen when charge finishes movement (this is a general player interrupt thing, no interrupt of any class has travel time)

**Comment by matanshukry** [comment]
Fixed

File: src/server/scripts/Spells/spell_druid.cpp [hunk 1]
```diff
@@ -1541,6 +1543,28 @@ class spell_dru_wild_growth : public SpellScriptLoader
...
 };
 
+// 106839 - Skull Bash
+class spell_dru_skull_bash : public SpellScript
+{
+    PrepareSpellScript(spell_dru_skull_bash);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_SKULL_BASH_CHARGE });
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp:1553]: Missing SPELL_DRUID_SKULL_BASH_INTERRUPT here

File: src/server/scripts/Spells/spell_druid.cpp [hunk 2]
```diff
@@ -1573,4 +1597,5 @@ void AddSC_druid_spell_scripts()
     new spell_dru_t10_restoration_4p_bonus();
     new spell_dru_t10_restoration_4p_bonus_dummy();
     new spell_dru_wild_growth();
+    RegisterSpellScript(spell_dru_skull_bash);
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp:1600]: sort by class name

File: src/server/scripts/Spells/spell_druid.cpp [hunk 3]
```diff
@@ -1541,6 +1543,28 @@ class spell_dru_wild_growth : public SpellScriptLoader
     }
 };
 
+// 106839 - Skull Bash
+class spell_dru_skull_bash : public SpellScript
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_druid.cpp:1547]: scripts are sorted (i dont remember why anymore, but they are)

**Review by Shauren — CHANGES_REQUESTED** [review]

**Review by Shauren — APPROVED** [review]

**Pull request merged**